### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by ZabihullahNooriWardak**
**In this branch, I have worked on the following tasks :** 

- [x] Implement class for Person
- [x] Implement class  for Student which inherit from Person 
- [ ] Implement class  for Teacher which inherit from Person 
- [ ] Implement encapsulation and inheritance with Ruby.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,126 @@
...
+
+To get a local copy up and running, follow these steps.
+
+1. Clone this repository to your local machine:
+
+```
+git clone https://github.com/ZabihullahNooriWardak/OOP-School-library.git
+
+```
+
```
Comment (tresorsawasawa) [inline README.md]: - [ **OPTIONAL** ] What do you think if you improve your `Getting Started` section of your README with other information needed by users in order to use your project locally(such as how to `access` the cloned dir, `use or run` the app, run tests if exixst,...) To help you, I recommend you use this **[Gist_file](https://gist.github.com/tresorsawasawa/0eba0570695aef068a7837c349f94737)** as template.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class Person
+  attr_accessor :id, :name, :age
+
+  def initialize(age, name: 'unknown', parent_permission: true)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
```
Comment (tresorsawasawa) [inline person.rb:8]: - [ ] As required for the project, kindly ensure that only the `Getter` is defined on the `:id` as it should only be able to be retrieved. Also you should ensure that it is present in the list of the **instance variable** such as in the example below.
```ruby
  ...
def initialize(age, name = "Unknown", parent_permission = true)
    @id = Random.rand(1..1000)
    @age = age
    @name = name
    @parent_permission = parent_permission
  end
  ...
```

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+  def private_of_age?
+    @age >= 18
+  end
+
+  def public_can_use_services?
+    private_of_age? || @parent_permission
+  end
```
Comment (tresorsawasawa) [inline person.rb:19]: - [ ] As required for the project, kindly ensure that the `#of_age?` method is set as `Private` method as right now all methods automatically `Public`. You can update them to look similar to this example below(or use another way to make it private):
```ruby
class Person
.
.
.
def initialize(age, name = "Unknown", parent_permission = true)
    @id = Random.rand(1..1000)
    @age = age
    @name = name
    @parent_permission = parent_permission
  end

  def can_use_services? 
    of_age?() || @parent_permission
  end

  private 

  def of_age?
    @age >= 18
  end
end
```

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @ZabihullahNooriWardak 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- Most of methods are correctly set up :heavy_check_mark: 
- Each class has its own file :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by divinecharlotte — APPROVED** [review]
# Hello @ZabihullahNooriWardak    😃 🥳 🎉 🎈 

Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 
### To Highlight 💯

- Good implementation of the required changes✔️
- Descriptive pr✔️






## Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**